File: lab3/shapes/src/TriangleSolution.java
public class TriangleSolution implements Shape {
    private int side1;
    private int side2;
    private int side3;
    public TriangleSolution(int side1In, int side2In, int side3In)
    {
        side1 = side1In;
        side2 = side2In;
        side3 = side3In;
    }

    @Override
    public String getShapeType()
    {
        return "Triangle";
    }

    @Override
    public int getPerimeter() {
        return side1 + side2 + side3;
    }

    @Override
    public int getArea() {
        return (int)Math.sqrt((side1 + side2 + side3) * (-side1 + side2 + side3) * (side1 - side2 + side3) * (side1 + side2 - side3) / 4);
    }
}

File: lab5-python/shapes/solution/triangle_solution.py
from shape import Shape
import math

class Triangle(Shape):
    def __init__(self, name, side1, side2, side3):
        super().__init__(name)
        self._side1 = side1
        self._side2 = side2
        self._side3 = side3

    def get_shape_type(self):
        return "Triangle"
    
    def get_perimeter(self):
        return self._side1 + self._side2 + self._side3
    
    def get_area(self):
        return math.sqrt( (self._side1  + self._side2 + self._side3) \
                        * (-self._side1 + self._side2 + self._side3) \
                        * (self._side1  - self._side2 + self._side3) \
                        * (self._side1  + self._side2 - self._side3)) / 4
